graphql/resolvers.go

```diff
@@ -2,22 +2,73 @@ package graphql
 
 import (
 	"context"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model/patch"
+	"github.com/evergreen-ci/evergreen/rest/data"
+	"github.com/pkg/errors"
 )
 
-type Resolver struct{}
+type Resolver struct {
+	sc data.Connector
+}
 
+func (r *Resolver) Patch() PatchResolver {
+	return &patchResolver{r}
+}
 func (r *Resolver) Query() QueryResolver {
 	return &queryResolver{r}
 }
 
+type patchResolver struct{ *Resolver }
+
+func (r *patchResolver) ID(ctx context.Context, obj *patch.Patch) (string, error) {
+	return obj.Id.Hex(), nil
+}
+func (r *patchResolver) Variants(ctx context.Context, obj *patch.Patch) ([]*string, error) {
+	builds := make([]*string, 0)
+	for _, b := range obj.BuildVariants {
+		builds = append(builds, &b)
+	}
+	return builds, nil
+}
+func (r *patchResolver) VariantTasks(ctx context.Context, obj *patch.Patch) ([]*VariantTask, error) {
+	variantTasks := []*VariantTask{}
+	for _, vt := range obj.VariantsTasks {
+		vtasks := make([]*string, 0)
+		for _, task := range vt.Tasks {
+			vtasks = append(vtasks, &task)
+		}
+		variantTasks = append(variantTasks, &VariantTask{
+			DisplayName: vt.Variant,
+			Tasks:       vtasks,
+		})
+	}
+	return variantTasks, nil
+}
+
 type queryResolver struct{ *Resolver }
 
-func (r *queryResolver) Test(ctx context.Context) (string, error) {
-	return "Hello World", nil
+func (r *queryResolver) UserPatches(ctx context.Context, userID string) ([]*patch.Patch, error) {
+	patches, err := r.sc.FindPatchesByUser(userID, time.Now(), 10)
```

Comment:
Let's actually change FindPatchesByUser to return the rest model, and move the conversion logic currently in https://github.com/evergreen-ci/evergreen/blob/master/rest/route/patch.go#L228 to this method

Reply:
assuming this will also be done later on when we figure out how to use the API models + when this code is more than placeholder